cpp/include/ucxx/endpoint.h

```diff
@@ -49,10 +49,11 @@ struct EpParamsDeleter {
  */
 struct ErrorCallbackData {
   Endpoint*
-    _endpoint;  ///< Pointer to the `ucxx::Endpoint` that owns this object, used only for logging.
+    endpoint;  ///< Pointer to the `ucxx::Endpoint` that owns this object, used only for logging.
   std::unique_ptr<std::mutex>
-    _mutex;  ///< Mutex used to prevent race conditions with `ucxx::Endpoint::setCloseCallback()`.
-  ucs_status_t status;                                 ///< Endpoint status
+    mutex;  ///< Mutex used to prevent race conditions with `ucxx::Endpoint::setCloseCallback()`.
+  ucs_status_t status;                         ///< Endpoint status
+  std::unique_ptr<std::atomic<bool>> closing;  ///< Prevent calling close multiple concurrent times.
```

Comment:
Yes, I did that to prevent having to add a constructor, but I now added it in https://github.com/rapidsai/ucxx/pull/184/commits/b7f09652031022539747bcadec6249b76c18fa0c .